packages/opencensus-exporter-jaeger/src/jaeger.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {classes, logger, types} from '@opencensus/opencensus-core';
+import * as os from 'os';
+
+import {Process, Tag} from './jaeger-driver/jaeger-thrift';
+import ThriftUtils from './jaeger-driver/thrift';
+import UDPSender from './jaeger-driver/udp_sender';
+import Utils from './jaeger-driver/util';
+
+/**
+ * Options for Jaeger configuration
+ */
+export interface JaegerTraceExporterOptions extends types.ExporterConfig {
+  serviceName: string;
+  tags?: Tag[];
+  host?: string;
+  port?: number;
+  maxPacketSize?: number;
+}
+
+
+/** Format and sends span information to Jaeger */
+export class JaegerTraceExporter implements types.Exporter {
+  // Name of the tag used to report client version.
+  static JAEGER_OPENCENSUS_EXPORTER_VERSION_TAG_KEY =
+      'opencensus.exporter.jaeger.version';
+  // host name of the process.
+  static TRACER_HOSTNAME_TAG_KEY = 'opencensus.exporter.jaeger.hostname';
+  //  ip of the process.
+  static PROCESS_IP = 'ip';
+
+  private process: Process;
+  private logger: types.Logger;
+  private sender: UDPSender;
+  checkCallback?: (exporter: JaegerTraceExporter) => void;
+  spanBuffer: types.Span[] = [];
+  successBuffer: types.Span[] = [];
```

Comment:
What is `successBuffer`? It looks like it is never getting emptied...

Reply:
successBuffer was replaced by successCount: number.